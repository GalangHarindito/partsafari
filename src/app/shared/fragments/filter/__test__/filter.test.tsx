import { render, screen, fireEvent } from '@testing-library/react';
import Filter from '../filter';
import { usePathname, useRouter, useSearchParams } from 'next/navigation';
import '@testing-library/jest-dom';

// Mock the useRouter hook
jest.mock('next/navigation', () => ({
  useRouter: jest.fn(),
  usePathname: jest.fn(),
  useSearchParams: jest.fn(),
}));

const mockPush = jest.fn();

describe('Filter Component', () => {
  beforeEach(() => {
    (useRouter as jest.Mock).mockReturnValue({ push: mockPush });
    (usePathname as jest.Mock).mockReturnValue('/test-path');
    (useSearchParams as jest.Mock).mockReturnValue(new URLSearchParams());
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  test('renders filter options correctly', () => {
    const data = [
      { id: 1, category: 'Category 1' },
      { id: 2, category: 'Category 2' },
    ];

    const { container } = render(<Filter data={data} />);

    expect(screen.getByText(/Filter:/)).toBeInTheDocument();
    expect(screen.getByText(/Category 1/i)).toBeInTheDocument();
    expect(screen.getByText(/Category 2/i)).toBeInTheDocument();
    expect(screen.getByText(/all/i)).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  test('sets active category on click', () => {
    const data = [
      { id: 1, category: 'Category 1' },
      { id: 2, category: 'Category 2' },
    ];

    render(<Filter data={data} />);

    const category1Chip = screen.getByText(/Category 1/i);
    fireEvent.click(category1Chip);

    expect(mockPush).toHaveBeenCalledWith('/test-path?tab=Category+1');
  });

  test('sets active category to "all" on initial render', () => {
    const data = [
      { id: 1, category: 'Category 1' },
      { id: 2, category: 'Category 2' },
    ];

    render(<Filter data={data} />);

    expect(mockPush).toHaveBeenCalledWith('/test-path?tab=all');
  });
});
