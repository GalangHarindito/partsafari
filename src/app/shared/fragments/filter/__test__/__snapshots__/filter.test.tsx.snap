// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter Component renders filter options correctly 1`] = `
<div>
  <div
    class="flex flex-row gap-3 items-center"
  >
    <p
      class="font-bold sm:text-sm text-xs"
    >
      Filter:
    </p>
    <div>
      <div
        class="p-4 py-2 rounded-full cursor-pointer bg-tab-active"
      >
        <p
          class="font-bold sm:text-sm text-xs"
        >
          ALL
        </p>
      </div>
    </div>
    <div>
      <div
        class="p-4 py-2 rounded-full cursor-pointer bg-tab-inactive"
      >
        <p
          class="font-bold sm:text-sm text-xs"
        >
          CATEGORY 2
        </p>
      </div>
    </div>
    <div>
      <div
        class="p-4 py-2 rounded-full cursor-pointer bg-tab-inactive"
      >
        <p
          class="font-bold sm:text-sm text-xs"
        >
          CATEGORY 1
        </p>
      </div>
    </div>
  </div>
</div>
`;
